cloudplatform/connectivity-apache-httpclient4/src/main/java/com/sap/cloud/sdk/cloudplatform/connectivity/HttpClientWrapper.java

```diff
@@ -76,6 +77,23 @@ public String toString()
     HttpClientWrapper( final CloseableHttpClient httpClient, final HttpDestinationProperties destination )
     {
         this.httpClient = httpClient;
+
+        // check destination is configured correctly
+        if( destination.get(DestinationProperty.PROXY_TYPE).contains(ProxyType.ON_PREMISE)
+            && destination.get(DestinationProperty.PROXY_HOST).isEmpty() ) {
+            final boolean isProvider =
+                destination.get(DestinationProperty.TENANT_ID).filter(String::isEmpty).isDefined();
+            final boolean isBasicAuth =
+                destination
+                    .get(DestinationProperty.AUTH_TYPE)
+                    .orElse(() -> destination.get(DestinationProperty.AUTH_TYPE_FALLBACK))
+                    .contains(AuthenticationType.BASIC_AUTHENTICATION);
+            if( isProvider && isBasicAuth ) {
+                throw new DestinationAccessException(
+                    "Accessing an onpremise system on behalf of the provider tenant explicitly is currently not supported.");
+            }
+            throw new DestinationAccessException("Unable to resolve connectivity service binding.");
```

Comment:
(Minor)

Maybe this is a good place to encourage looking into the logs?

```suggestion
            throw new DestinationAccessException("Unable to resolve connectivity service binding. Please check the logs.");
```

Reply:
I stored the failed proxy destination and re-throwing it in the HttpClientWrapper. Tell me what you think.